sdk/python/feast/feature_server.py

```diff
@@ -201,13 +201,22 @@ def push(body=Depends(get_body)):
         for feature_view in fvs_with_push_sources:
             assert_permissions(resource=feature_view, actions=actions)
 
-        store.push(
+        push_params = dict(
             push_source_name=request.push_source_name,
             df=df,
             allow_registry_cache=request.allow_registry_cache,
             to=to,
         )
 
+        should_push_async = (
+            store._get_provider().async_supported.online.write
+            and to in [PushMode.ONLINE, PushMode.ONLINE_AND_OFFLINE]
+        )
+        if should_push_async:
+            await store.push_async(**push_params)
+        else:
+            store.push(**push_params)
```

Comment:
if async writes to the online store are supported and the post body indicates that data will be written to the online store, then attempt an `async storep.push`